presto-main/src/main/java/com/facebook/presto/sql/gen/ExpressionProfiler.java

```diff
@@ -14,66 +14,57 @@
 package com.facebook.presto.sql.gen;
 
 import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Ticker;
 
+import static com.facebook.presto.execution.executor.PrioritizedSplitRunner.SPLIT_RUN_QUANTA;
+import static com.google.common.base.Ticker.systemTicker;
 import static com.google.common.base.Verify.verify;
+import static java.util.Objects.requireNonNull;
 import static java.util.concurrent.TimeUnit.NANOSECONDS;
 
 public class ExpressionProfiler
 {
-    private static final int NUMBER_OF_ROWS_TO_PROFILE = 1024;
-    private static final int EXPENSIVE_FUNCTION_THRESHOLD_MILLIS = 1_000;
+    private static final long EXPENSIVE_EXPRESSION_THRESHOLD_MILLIS = SPLIT_RUN_QUANTA.toMillis();
     private static final int NOT_INITALIZED = -1;
 
-    private final int rowsToProfile;
-    private final int expensiveFunctionThresholdMillis;
-    private double meanExecutionTime;
+    private final Ticker ticker;
+    private final long expensiveExpressionThresholdMillis;
+    private double totalExecutionTime;
     private int samples;
     private long previousTimestamp = NOT_INITALIZED;
     private boolean isExpressionExpensive = true;
-    private boolean isProfiling = true;
 
     public ExpressionProfiler()
     {
-        this.rowsToProfile = NUMBER_OF_ROWS_TO_PROFILE;
-        this.expensiveFunctionThresholdMillis = EXPENSIVE_FUNCTION_THRESHOLD_MILLIS;
+        this.expensiveExpressionThresholdMillis = EXPENSIVE_EXPRESSION_THRESHOLD_MILLIS;
+        this.ticker = systemTicker();
```

Comment:
You can use a passthrough constructor call here `this(systemTicker(), EXPENSIVE_EXPRESSION_THRESHOLD_MILLIS)`